### PR TITLE
Update `PopupRoute` docs and add an example

### LINKED ISSUE #106650: `PopupRoute`: Document dismiss action behavior and add an example
Dismiss action behavior added in https://github.com/flutter/flutter/pull/59310  is not documented. 

When `PopupRoute` is on the screen, pressing the `ESC` key dismisses the popup route. This is not mentioned in the docs, only testing the code reveals this behavior.

related https://github.com/flutter/flutter/pull/106186

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/106650

related https://github.com/flutter/flutter/pull/106186

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/widgets/routes/popup_route.0.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: PopupRoutExample(),
+    );
+  }
+}
+
+class PopupRoutExample extends StatelessWidget {
```
Comment (Piinks) [inline examples/api/lib/widgets/routes/popup_route.0.dart:22]: Typo
```suggestion
class PopupRouteExample extends StatelessWidget {
```

File: packages/flutter/lib/src/widgets/routes.dart [hunk 2]
```diff
@@ -1691,6 +1691,22 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 }
 
 /// A modal route that overlays a widget over the current route.
+///
+/// When [barrierDismissible] is true, the user can dismiss the route by tapping
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/routes.dart:1695]: IT looks like this is documented on ModalRoute.barrierDismissible, can you update those docs and create a template to re-use here?

**Review by Piinks — COMMENTED** [review]
Nice samples! Thanks for adding those tests to the other one that already existed! :)

File: packages/flutter/lib/src/widgets/routes.dart (hunk 2 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/widgets/routes.dart]: Great idea! I just updated ModalRoute docs, I noticed docs for `ModalRoute.barrierDismissible` is a template itself but it has too many details that we don't need here, and turned the part we need into a smaller template and mentioned macro here.

Comment (Piinks) [reply packages/flutter/lib/src/widgets/routes.dart]: Fab!

File: packages/flutter/lib/src/widgets/routes.dart [hunk 3]
```diff
@@ -1691,6 +1694,18 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
...
+///
+/// {@tool dartpad}
+/// This example shows how to create a dialog box that is dismissible.
+///
+/// ** See code in examples/api/lib/widgets/routes/popup_route.0.dart **
+/// {@end-tool}
+///
+/// See also:
+/// * [ModalRoute], which is the base class for this class.
+/// * [Navigator.pop], which is used to dismiss the route.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/routes.dart:1708]:
```suggestion
/// See also:
///
///   * [ModalRoute], which is the base class for this class.
///   * [Navigator.pop], which is used to dismiss the route.
```

**Review by Piinks — APPROVED** [review]
LGTM with nit. Thank you!

File: packages/flutter/lib/src/widgets/routes.dart (hunk 2 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/widgets/routes.dart]: Actually, I had to revert the smaller template as it was nested, after updating the PR, the analyzer throws.
So I simply added the entire `ModalRoute.barrierDismissible` template to PopupRoute docs

**Pull request merged**